pkg/R/utils.R

```diff
@@ -164,11 +176,11 @@ reserialize <- function(rdd) {
   if (!inherits(rdd, "RDD")) {
     stop("Argument 'rdd' is not an RDD type.")
   }
-  if (rdd@env$serialized) {
-    return(rdd)
-  } else {
+  if (rdd@env$serialized == "string") {
```

Comment:
Agreed. I've updated `reserialize` to account for this, like so:

```R
reserialize <- function(rdd) {
  if (!inherits(rdd, "RDD")) {
    stop("Argument 'rdd' is not an RDD type.")
  }
  if (rdd@env$serializedMode == "string") {
    ser.rdd <- lapply(rdd, function(x) { x })
    return(ser.rdd)
  } else if (rdd@env$serializedMode == "row") {
    rdd@env$serializedMode == "byte"
    ser.rdd <- lapply(rdd, function(x) { x })
    return (ser.rdd)
  } else {
    return(rdd)
  }
}
```
This way, things like `UnionRDD` can handle mixed serialization types and will even return the column names in the `row` serialized part of the union.